crates/network/src/client.rs

```diff
@@ -63,7 +63,7 @@ pub type DatagramHandlers = HashMap<u32, (&'static str, DatagramHandler)>;
 /// Represents either side of a high level connection to a game client of some sort.
 ///
 /// Allows making requests and RPC, etc
-pub trait ClientConnection: 'static + Send + Sync {
+pub trait NetworkTransport: 'static + Send + Sync {
```

Comment:
Renamed `ClientConnection` to `NetworkTransport`.

This is used by both the server and the client to abstract over a kind of network connection. It abstracts over quinn, but also a proxy for the webtransport driver. `WebTransportSession` is not send, and therefore the message queue proxy has to be used instead. Hence, it isn't really a *Connection* trait, but rather a *Transport* for a connection. Besides, *Connection* is already imported in scope in many files, most commontly `quinn::Connection`, which is wrapped by our ambient proxy `ConnectionKind` and *then* becomes a transport.

This makes the naming a lot clearer and avoids using *Client...* inappropriately, and also avoids just remixing the same words into different types.